tasks/SingleCapturer.cpp

```diff
@@ -0,0 +1,78 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "SingleCapturer.hpp"
+
+#include "frame_helper/FrameHelper.h"
+
+using namespace video_streamer_vlc;
+
+SingleCapturer::SingleCapturer(std::string const& name)
+    : SingleCapturerBase(name)
+{
+}
+
+SingleCapturer::~SingleCapturer()
+{
+}
+
+
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See SingleCapturer.hpp for more detailed
+// documentation about them.
+
+bool SingleCapturer::configureHook()
+{
+    if (! SingleCapturerBase::configureHook())
+        return false;
+    
+    if (_input_buffer_ms.value() >=0 ) {
+        capturer = std::make_shared<VlcCapture>(_uri.value(), _input_buffer_ms.value());
+    } else {
+        capturer = std::make_shared<VlcCapture>(_uri.value());
+    }
+    capturer->callbacks.push_back(this);
+
+    return true;
+}
+bool SingleCapturer::startHook()
+{
+    if (! SingleCapturerBase::startHook())
+        return false;
+    capturer->start();
+    return true;
+}
+void SingleCapturer::updateHook()
+{
+    SingleCapturerBase::updateHook();
+
+    cv::Mat image;
+    if (capturer->read(image)) {
+
+        cv::Mat newmat(image.size(), CV_8UC3);
+        cv::cvtColor(image, newmat, cv::COLOR_RGBA2BGR);
```

Comment:
While using this with cameras in MODKOM, we found an issue with this color conversion. The `SingleStreamer.cpp` seems to convert everything into RGB, but this snippet converts that into BGR, resulting in wrong colors.

For us works this change:

```suggestion
        cv::cvtColor(image, newmat, cv::COLOR_RGBA2RGB);
```

Reply:
Hmm, the ```frame_helper::FrameHelper::copyMatToFrame(newmat,frame);``` function used below only supports BGR and fails for RGB:  https://github.com/rock-core/perception-frame_helper/blob/master/src/FrameHelper.cpp#L1114C23-L1114C37

It could be that the in the capturer "RV24" is not always RGBA, in this line: https://github.com/rock-drivers/drivers-video_capture_vlc/blob/master/src/VlcCapture.cpp#L94. then we'd need to adapt the source formar accordingly. Or the error is somewhere in the following code

Reply:
Actually is seems to be an issue that the `frame_helper` always interprets `CV_8UC3` as being `BGR`. However in the OpenCV documentation is it explicitly stated, that this is not the case and the order of color values has to be specified elsewhere.
See also: https://stackoverflow.com/questions/27183946/what-does-cv-8uc3-and-the-other-types-stand-for-in-opencv